### PR TITLE
Run number

### PR DESCRIPTION
**Pull request by dneise**
extract the `run_id` from the `sfc.main_page().system_status` string and expose as `sfc.main_page.run_id`

c.f. https://github.com/fact-project/la_palma_overview/pull/7

### CODE REVIEW

**Comment by MaxNoe** [comment]
Sorry, too fat fingers and to small cell phone buttons.

Why a string and not an integer?

**Comment by dneise** [comment]
Because same return type in both cases, existing run_id and non-existent run_id in system_status. 
User does not have to check for None.

**Comment by MaxNoe** [comment]
But that carries information. I would opt for returning an integer or `None`

**Comment by relleums** [comment]
I also suggest to always return a string. First, our so far only use case is only about showing the run-string to a user and not doing any calculations where an integer would be needed. Second, although cumbersome the user still has the chance to make an integer out of the string. I am afraid we can not provide or guarantee the user to have always nice representations when we have to code against this smartfact interface. The effort to explore the braces and foo in this string is not worth risking exceptions raising in the smart fact crawler to provide the run-id as an int and thus risking smooth operation at night.

**Comment by MaxNoe** [comment]
If this is called run_NUMBER i expect a number. Period.

**Comment by dneise** [comment]
Ah ... so either we call it 'run' then .... or even better 'the_text_that_happens_to_be_between_the_brackets_if_there_are_any_else_None'

**Comment by dneise** [comment]
... ah ... Period as well ;-)

**Comment by MaxNoe** [comment]
Sorry for the period.

File: smart_fact_crawler/__init__.py [hunk 1]
```diff
@@ -266,18 +256,19 @@ def main_page(url=None, timeout=None):
         humidity = Quantity(float('nan'), '%')
         wind_speed = Quantity(float('nan'), 'km/h')
 
-
+    run_id_in_system_status_pattern = r'.*\((\d*)\).*'
```
Comment (MaxNoe) [inline smart_fact_crawler/__init__.py]: If we change `\d*` to `\d+` it's always convertible to an integer if it matches

File: smart_fact_crawler/__init__.py [hunk 2]
```diff
@@ -266,18 +256,19 @@ def main_page(url=None, timeout=None):
...
+    match_run_id = re.match(run_id_in_system_status_pattern, table[1][1])
     return to_namedtuple('MainPage', {
         'timestamp_1': sft2dt(table[0][0]),
         'timestamp_2': sft2dt(table[0][1]),
         'system_status': table[1][1],
         'relative_camera_temperature': Quantity(s2f(table[3][1]), 'deg_C'),
         'humidity': humidity,
         'wind_speed': wind_speed,
+        'run_id': match_run_id.groups()[0] if match_run_id is not None else '',
```
Comment (MaxNoe) [inline smart_fact_crawler/__init__.py]: Our run_id is an integer, I would like it to be an integer here as well.

File: smart_fact_crawler/__init__.py (hunk 1 above)
Comment (dneise) [inline smart_fact_crawler/__init__.py]: Ah, right ... I had a `.*` there before and still had that in mind. 
Right so this is then a safe conversion. Cool

**Comment by dneise** [comment]
It will still throw a `TypeError`, when the result of `table[1][1]` is not string-like enough to be used inside `re.match()`, but this was the case independ of the cast-to-int question. 
We could swallow this exception silently if we want to avoid this to cause a developer call, or we could hide the exception and return None, but log it. So at least we see it in the logs (when ever somebody actually has a look) ... 

What do you think about this?

**Comment by dneise** [comment]
c.f. https://github.com/fact-project/la_palma_overview/pull/7

**Comment by MaxNoe** [comment]
How can table[1][1] not be a string? Either it is a string or it throws an index error, or am I mistaken?

**Comment by dneise** [comment]
It can only be a string at the moment, but there is no test-harness to ensure it.

**Pull request merged**